packages/service-library/src/servicelib/logging_utils.py

```diff
@@ -121,6 +136,20 @@ def set_logging_handler(
         )
 
 
+class GeneralLogFilter(logging.Filter):
+    def __init__(self, filtered_routes):
+        super().__init__()
+        self.filtered_routes: list[str] = filtered_routes
+
+    def filter(self, record):
+        msg = record.getMessage()
+
+        # Check if the filtered routes exists in the message
+        return not any(
+            filter_criteria in msg for filter_criteria in self.filtered_routes
```

Comment:
You may think this is a premature overoptimization but here I would really suggest to go for regexes right away. They are meant for this. You can easily validate if some input string that claims to be a regex is actually a regex. Everyone can lookup regexes on their own if log whitespacing changes or so, and optimize.

Reply:
I mean, match input regexes with the loglines instead of using python's `in`

Reply:
Or globbing would also be nice, in case you somehow dont like regex. I'd prefer regexes over globbing tho.

Reply:
I do agree that regex might be a better one, at least from usage point of view. although they are maybe slower than straight string matching. And as I said I am wondering a bit on the performance impact of that thing.

Reply:
The `in` operator is generally faster when you need to check if a substring exists within a string and it's simple python approach. Regular expressions are slower than `in` because they offer more flexibility and need to compile and process the pattern to perform the search. I think in this use case using `in` is better option. Also it doesn't prevent us in the future to change it to regex if needed.

Reply:
https://github.com/ITISFoundation/osparc-simcore/pull/6596#discussion_r1815219159 Yes performance will be observed.

Reply:
what a `GeneralLogFilter` has to do with `filtered_routes`?

You have a list of strings and want to make sure that if the log message include any of these strings, it gets filtered out.


```python
class ExcludeKeywordsLogFilter(logging.Filter)
    def __init__(self, keywords: set[str])
        super().__init__()
        self.excluded_keywords = keywords

    def filter(self, record):
        log_msg = record.getMessage()
        found = any( keyword in log_msg for keyword in self.excluded_keywords )
        return not found
```

Whether to use regex or not, i think this you have to really measure it. My bet is that this method is faster for this particular use case

Reply:
I actually recommend a test (e.g. using pytest-benchmark) that keeps in check the timing with and without filter